core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -685,7 +685,7 @@ private[ui] class TaskDataSource(
 
   private var _tasksToShow: Seq[TaskData] = null
 
-  override def dataSize: Int = taskCount(stage)
+  override def dataSize: Int = store.taskCount(stage.stageId, stage.attemptId).toInt
```

Comment:
nit: after this change, the function `taskCount()` is only referenced by `totalTasks`, we can inline that.

Reply:
Thanks. Done.